tests/test_users.py

```diff
@@ -58,154 +57,153 @@ class TestUser:
     """ A collection of tests of the User part of the DB.
     """
 
-    def test_add_user(self, session, user_json_object):
+    def test_add_user(self, client, user_json_object):
         """ Test adding a new user to the DB """
         user = user_json_object  # Convenience renaming
 
-        new_user = hydra_base.add_user(user)
-
+        new_user = client.add_user(user)
         assert new_user.username == user.username, "Usernames are not the same!"
-
-        assert bcrypt.checkpw(user.password.encode('utf-8'), new_user.password)
+        #convert pwd to bytes if it's not bytes
+        pwd_hash = new_user.password if isinstance(new_user.password, bytes) else new_user.password.encode('utf-8')
```

Comment:
So here the `.password` attr on the `new_user` returned from `client.add_user` is actually a _hash_ of the password if it can be used as the second arg to `bcrypt.checkpw`.

Somewhat confusing name, perhaps something indicating that it is a hash would be clearer.

Reply:
This would require renaming the database column and all associated variables. I'm not sure it's worth it. If you feel strongly about it, I can raise an issue to deal with it explicitly

Reply:
I agree that's too much change for a minor issue like this, also with consequences elsewhere.  Perhaps an explanatory comment that it is the password hash being converted to bytes.